runners/flink/build.gradle

```diff
@@ -134,5 +134,31 @@ task validatesRunner {
   dependsOn validatesRunnerStreaming
 }
 
+task runJobServer(type: JavaExec) {
+  classpath = sourceSets.main.runtimeClasspath
+  main = "org.apache.beam.runners.flink.FlinkJobServerDriver"
+  def jobHost = project.hasProperty("jobHost") ? project.property("jobHost") : "localhost:8099"
+  def artifactsDir = project.hasProperty("artifactsDir") ?  project.property("artifactsDir") : "/tmp/flink-artifacts"
+  args = ["--job-host=${jobHost}", "--artifacts-dir=${artifactsDir}"]
+  // Enable remote debugging.
+  jvmArgs = ["-Xdebug", "-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005"]
+}
+
+task jobServerShadowJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar) {
```

Comment:
It's needed in order to make the job server submittable to a Flink cluster via `flink run`. If we already had an executable uberjar that we publish that would be fine.

Reply:
It seems like we should publish both the old Flink artifact which people use right now and this job server artifact with Apache Beam. The easiest way to do this is to create a separate job-server subproject in a directory below the Flink directory and produce this as an artifact.

Reply:
If we have the uber flink jar then we can use it as it is.
What is the command to create the uber jar in gradle?

Reply:
@lukecwik I think that makes sense. I'm not sure how to reach into a different project's configurations in order to construct the dependency set though.

Reply:
I would think that this would add the project and all its dependencies to the project dependency list:
```
configurations {
  compile project(path: "beam-runners-flink", configuration: "shadow")
}
```

Reply:
I ended up doing it in a different way. Let me know what you think.

Reply:
The main downside of my approach is that it requires an evaluation dependency.

Reply:
@angoenka With the new layout, to create the uber jar, you run `./gradlew -p runners/flink/job-server shadowJar`. It will output the uber jar at `runners/flink/job-server/build/libs/flink-job-server.jar`. Note that if we plan to actually publish this, we may want to go back to a versioned artifact name.

In order to debug the job server through gradle: `./gradlew -p runners/flink/job-server runShadow -PjobHost=<server address> -PartifactsDir=<artifact staging directory>`.